popde/density_estimate.py

```diff
@@ -1,3 +1,71 @@
+import numpy as np
+
+class SimpleKernelDensityEstimation:
+    """
+    Fit and evaluate multi-dimensional Kernel Density Estimation (KDE) 
+
+    Methods:
+    --------
+    check_dimensionality():
+        check if the data matches the dimensions of KDE.
+
+    evaluate(points):
+        Evaluate the KDE at given data points.
+    """
+    def __init__(self, data, bandwidth=1.0, kernel='gaussian', dim_names=None):
+        """
+        Initialize the KernelDensityEstimation object.
+        data: array-like, shape (n_samples, n_features)
+               points of the data define each kernel position
+               each row is a point, each column is a parameter.
+        kwargs:
+            dim_names : sequence of dimension names, e.g. ('m1', 'z', 'chi_eff') 
+                        values must be strings
+        """
+        if len(data.shape) != 2:
+            raise ValueError("data must have shape (n_samples, n_features).")
+
+        self.data = np.asarray(data) 
+        self.bandwidth = bandwidth
+        self.kernel = kernel 
+
+        self.dim_names = dim_names
+        if dim_names is not None:
+            self.check_dimensionality()
+
+    def check_dimensionality(self):
+        """
+        Check if the dimension of KDE matches the parameters on which we evaluate KDE.
+        Raises ValueError
```

Comment:
It is obvious from the code that it raises a ValueError, no need for an extra comment line
```suggestion
        Check if the dimension of training data matches the number of parameter names.
```

Reply:
fixed